bc_obps/reporting/tests/service/test_compliance_service/test_compliance_service_class.py

```diff
@@ -55,18 +61,77 @@ def test_compliance_summary_with_all_data(self):
         assert result.excess_emissions == Decimal('49232.7564')
         assert result.credited_emissions == 0
 
+    def test_apr_dec_used_for_2024_summary(self):
+        # Explicit test that 2024 uses Apr-Dec scaled allocations for compliance
+        build_data = ComplianceTestInfrastructure.build()
+        result = ComplianceService.get_calculated_compliance_data(build_data.report_version_1.id)
+
+        assert result.emissions_attributable_for_reporting == Decimal('123001.0577')
+        assert result.reporting_only_emissions == Decimal('3000.05')
+        # Apr–Dec scaled allocated total for 2024
+        assert result.emissions_attributable_for_compliance == Decimal('70000.2564')
+        assert result.emissions_limit == Decimal('20767.5000')
+        assert result.excess_emissions == Decimal('49232.7564')
+        assert result.credited_emissions == Decimal('0')
+
+    def test_apr_dec_product_level_behavior(self):
+        """Per-product Apr–Dec allocation behavior test."""
+        # Build 2024 data
+        build_2024 = ComplianceTestInfrastructure.build()
+        result_2024 = ComplianceService.get_calculated_compliance_data(build_2024.report_version_1.id)
+
+        # Create a 2025 report from the same base data
+        build_2025 = ComplianceTestInfrastructure.reporting_year_2025()
+        result_2025 = ComplianceService.get_calculated_compliance_data(build_2025.report_version_1.id)
+
+        report_version_id = build_2024.report_version_1.id
+        basic_category_ids = list(EmissionCategory.objects.filter(category_type="basic").values_list("id", flat=True))
+
+        for prod_2024 in result_2024.products:
+            prod_2025 = next((p for p in result_2025.products if p.product_id == prod_2024.product_id), None)
+            self.assertIsNotNone(prod_2025, f"Product {prod_2024.product_id} missing in 2025 result")
+
+            # compute full-year allocated (basic - reporting-only)
+            basic_allocated = ComplianceService.get_allocated_emissions_by_report_product_emission_category(
+                report_version_id, prod_2024.product_id, basic_category_ids
+            )
+            reporting_only = ComplianceService.get_reporting_only_allocated(report_version_id, prod_2024.product_id)
+            allocated_full = Decimal(basic_allocated) - Decimal(reporting_only)
+
+            # production totals
+            totals = ComplianceService.get_report_product_aggregated_totals(report_version_id, prod_2024.product_id)
+            annual_amount = Decimal(totals.get("annual_amount") or 0)
+            apr_dec_amount = Decimal(totals.get("apr_dec") or 0)
+
+            # expected values for 2024 (Apr-Dec scaling if annual_amount > 0)
+            if annual_amount == 0:
+                expected_2024 = Decimal("0.0000")
+            else:
+                expected_2024 = (allocated_full / annual_amount) * apr_dec_amount
```

Comment:
This value should be pre-calculated and hard-coded in this test. Any change in the data input should break this test